app/src/main/java/org/hackillinois/android/view/home/HomeFragment.kt

```diff
@@ -0,0 +1,112 @@
+package org.hackillinois.android.view.home
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.widget.LinearLayoutManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.fragment_home.*
+import kotlinx.android.synthetic.main.fragment_home.view.*
+import org.hackillinois.android.R
+import org.hackillinois.android.model.EventsList
+import org.hackillinois.android.utils.StartTimes
+import org.hackillinois.android.utils.TimeInfo
+import org.hackillinois.android.viewmodels.home.HomeViewModel
+import java.util.*
+
+class HomeFragment : Fragment(), CountdownManager.CountDownListener {
+
+    private val eventsAdapter = EventsListAdapter(mutableListOf())
+
+    private val eventTimes = listOf(StartTimes.eventStartTime, StartTimes.hackingStartTime, StartTimes.hackingEndTime)
+    private val titles = listOf("EVENT STARTS IN", "HACKING STARTS IN", "HACKING ENDS IN", "THANKS FOR COMING!")
+
+    private lateinit var viewModel: HomeViewModel
+    private val countDownManager = CountdownManager(this, eventTimes)
+
+    private var title = "Title"
```

Comment:
I'm a little confused what this is used for

Reply:
`this` or `title`?

Reply:
`this` is used so that countdownManager can send data back to HomeFragment without knowing that its a fragment

`title` is used because the title is sent back from `countdownManager` before the views are created, so the first title is always incorrect unless we store it in a variable and use that variable in `onCreateView`. I may change this so that `countdownManager` is started after the view is created.